query-compiler/query-compiler/src/translate.rs

```diff
@@ -101,24 +108,133 @@ impl<'a, 'b> NodeTranslator<'a, 'b> {
         Ok(children)
     }
 
+    fn wrap_children_with_expr(&self, expr: Expression, children: Vec<Expression>) -> Expression {
+        if children.is_empty() {
+            return expr;
+        }
+        Expression::Let {
+            bindings: vec![Binding::new(self.node.id(), expr)],
+            expr: if children.len() == 1 {
```

Comment:
nitpick but I like `exactly_one` for stuff like this
```suggestion
            expr: children.into_iter().exactly_one().unwrap_or_else(Expression::Seq).into()
```

Reply:
I can change this if you feel strongly, but fwiw it will lead to an extra allocation because the `exactly_one` error is an iterator and not the original collection:
```rust
                Ok(children
                    .into_iter()
                    .exactly_one()
                    .unwrap_or_else(|e| Expression::Seq(e.collect()))
                    .into())
```

Reply:
fair enough